src/App.tsx

```diff
@@ -8,21 +8,123 @@ import { ShortTitleCard } from "./components/molecules/short-title-card";
 import { Title } from "./components/atoms/Title";
 import { TrailerCard } from "./components/molecules/trailer-card";
 import { StarRating } from "./components/molecules/star-rating";
-import { FilterPage } from "./components/molecules/filter-page";
+import { FilterPage } from "./components/organism/filter-page";
 import { movie, trailer } from "./mock";
 import { useState } from "react";
 import { Filter } from "./components/atoms/Filter";
+import { ISettingsSort } from "./types";
+import { ButtonShowFilm } from "./components/atoms/ButtonShowFilm";
 
 function App() {
-  const selectedUser = movie[1];
+  /* const selectedUser = movie[1]; */
   const trailerVideo = trailer;
-  const [filteredFilms, setFilteredFilms] = useState(movie);
+  const countries = movie.map((movie) => movie.country);
+  /*  const countries = массивуникальных ингредиентов, 
+  прокинуть в фильтр пэйдж сантрис = кантрис через пропсы , по массиву пройти map ,
+  onChance  и через target value*/
+
+  //нажатие на маленькую карточку- выводим большую с этим же id
+  const onClickFilm = (id: number) => {
+    console.log({ id });
+    const newFilm = movie.find(({ id: filmId }) => id === filmId);
+
+    if (newFilm) {
+      setSelectedFilm(newFilm);
+    }
+  };
+
+  //add закладку
+  const addBookmark = (id: number) => {
+    console.log("addBookmark");
+    const hasId = bookmarksId.find((currentId) => currentId === id);
+    if (hasId) {
+      return;
+    }
+    const newBookmark = [...bookmarksId, id];
+    setBookmarksId(newBookmark);
+    localStorage.setItem("bookmarks", JSON.stringify(newBookmark));
+  };
+
+  //remove закладку
+  const removeBookmark = (id: number) => {
+    console.log("setBookmarksId");
+    const filtredBookmarks = bookmarksId.filter(
+      (currentId) => currentId !== id
+    );
+    setBookmarksId(filtredBookmarks);
+    localStorage.setItem("bookmarks", JSON.stringify(filtredBookmarks));
+  };
+
+  // подключение localStorage к закладкам
+  useEffect(() => {
+    const savedBookmarks = localStorage.getItem("bookmarks");
+    if (savedBookmarks) {
+      setBookmarksId(JSON.parse(savedBookmarks));
+    }
+    const savedViewedFilms = localStorage.getItem("viedFilm");
+    if (savedViewedFilms) {
+      setViewedFilm(JSON.parse(savedViewedFilms));
+    }
+
+    return () => {
+      //аналог анмаунта
+    };
+  }, []);
+
+  // Ставим метку просмотрено
+  const onChangeSwitcher = (id: number, checked: boolean) => {
+    setIsViewedChecked(!isViewedChecked);
+
+    const newViewedFilm = checked
+      ? viewedFilm.filter((currentId) => currentId !== id)
+      : [...viewedFilm, id];
+
+    setViewedFilm(newViewedFilm);
+
+    localStorage.setItem("viedFilm", JSON.stringify(newViewedFilm));
+  };
+
+  const [viewedFilm, setViewedFilm] = useState<number[]>([]);
+  const [isViewedChecked, setIsViewedChecked] = useState(false);
```

Comment:
const [isViewedChecked, setIsViewedChecked] = useState(false);
are you sure this code is needed?